### DISCUSSION TITLE
Correct way to spawn sprite

### DISCUSSION THREAD

**Discussion by bfg-coding** [bevyengine/bevy#9190] [Q&A] [answered]
I just finished a Bevy 0.10 tutorial series and wanted to apply my knowledge and failed instantly on trying to spawn my player like I did in the tutorial.  What is the most correct way to spawn something into the engine for 2D games.  Code below for starting point

```rust
fn spawn_player(
    mut commands: Commands,
    window_query: Query<&Window, With<PrimaryWindow>>,
    asset_server: Res<AssetServer>
) {
    let window = window_query.get_single().unwrap();
    commands.spawn(
        (
            SpriteBundle {
                transform: Transform::from_xyz(window.width() / 2.0, window.height() / 2.0, 0.0),
                texture: asset_server.load("sprites/player/player.png"),
                ..default()
            },
            Player {}
        )
    );
}
```

**Comment by mockersf** [comment, marked as answer]
I'm not sure why you're using the window to move your sprite. I think that transform would move it just at the border of the screen.
You also need to spawn a camera.

You can look at the [sprite example](https://github.com/bevyengine/bevy/blob/latest/examples/2d/sprite.rs) for how to spawn a sprite. 
https://github.com/bevyengine/bevy/blob/8ba9571eedada4f3ff43cdf1402670b7fe7c280d/examples/2d/sprite.rs#L12-L18

**Reply by bfg-coding** [reply]
I had the camera in a different function.  Thanks for the link I'll check it out
